lib/commands/run.coffee

```diff
@@ -135,22 +135,27 @@ maybeInspectContainer = (docker, name) ->
 # Helper method to check and see if the running container's links differ from the ones we would
 # want to create it with. If they do differ, the only recourse is to remove and recreate the
 # container, since you may not (currently) modify links of a running container.
-isLinkMissing = (containerInfo, createOpts) ->
-  # These are the links as reported back by Docker, which have the format:
-  #  /sourceContainer:/destContainer/alias
-  # We do a bit of splitting and joining to convert them back to /sourceContainer:alias so that
-  # we can compare them directly with the Links parameter we provide to create.
-  currentLinks = _.map (containerInfo?.HostConfig?.Links or []), (link) ->
-    [source, dest] = link.split(':')
-    "#{source}:#{dest.split('/').pop()}"
+isLinkMissing = (containerInfo, createOpts, servicesConfig) ->
+  service = containerInfo.Name.split("/")[1].split(".")[0]
+  env = containerInfo.Name.split("/")[1].split(".")[1]
+  prereqs = ServiceHelpers.generatePrereqServices(service, servicesConfig)
+  docker = new Docker()
+  ops = prereqs.map (prereq) ->
+    prereqContainerName = "#{prereq}.#{env}"
+    DockerUtils.inspectContainer docker.getContainer(prereqContainerName)
+    .then ({container, info}) ->
+      if info.Created > containerInfo.Created
+        console.log "the link to", info.Name, " is broken, recreate"
+        true
+      else
+        false
 
-  # Make a copy since sort mutates.
-  requestedLinks = createOpts.HostConfig.Links.concat()
+  RSVP.all _.compact(ops)
+  .then (results) ->
+    for result in results
+      console.log "result: ", result
```

Comment:
That should be fine, but just bear in mind that `isLinkMissing` now needs to return a `Promise` in all cases because it's now async. That means that `containerNeedsRecreate` needs to change to return a `Promise`, and so on from there.